components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/ApplicationCertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,196 @@
+        }
+        Certificate certificate = DAO.getCertificateByName(certificateName,
+                IdentityTenantUtil.getTenantId(tenantDomain));
+        if (certificate == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No certificate found for the name: " + certificateName);
+            }
+            throw CertificateMgtUtil.raiseClientException(
+                    CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST_WITH_GIVEN_NAME, certificateName);
+        }
+
+        return certificate;
+    }
+
+    /**
+     * Update a certificate with given id.
+     *
+     * @param certificateId      Certificate ID.
+     * @param certificateContent Certificate content.
+     * @param tenantDomain       Tenant domain.
+     * @throws CertificateMgtException If an error occurs while updating the certificate.
+     */
+    @Override
+    @Deprecated
+    public void updateCertificateContent(int certificateId, String certificateContent, String tenantDomain)
+            throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating certificate with id: " + certificateId);
+        }
+        checkIfCertificateExists(certificateId, tenantDomain);
+        CERTIFICATE_VALIDATOR.validateCertificateContent(certificateContent);
+        DAO.updateCertificateContent(certificateId, certificateContent, IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    /**
+     * Delete a certificate with given id.
+     *
+     * @param certificateId Certificate ID.
+     * @param tenantDomain  Tenant domain.
+     * @throws CertificateMgtException If an error occurs while deleting the certificate.
+     */
+    @Override
+    @Deprecated
+    public void deleteCertificate(int certificateId, String tenantDomain) throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Deleting certificate with id: " + certificateId);
+        }
+        DAO.deleteCertificate(certificateId, IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    /**
+     * Validate the certificate.
+     *
+     * @param certificate Certificate information.
+     */
+    private void doPreAddCertificateValidations(Certificate certificate) throws CertificateMgtClientException {
+
+        CERTIFICATE_VALIDATOR.validateForBlank(CertificateValidator.NAME_FIELD, certificate.getName());
+        CERTIFICATE_VALIDATOR.validateForBlank(CertificateValidator.CERTIFICATE_FIELD, certificate.getCertificate());
+        CERTIFICATE_VALIDATOR.validatePemFormat(certificate.getCertificate());
+    }
+
+    /**
+     * Check if the certificate existence.
+     *
+     * @param certificateId     Action ID.
+     * @param tenantDomain Tenant Domain.
+     * @throws CertificateMgtException If the certificate does not exist.
+     */
+    private void checkIfCertificateExists(int certificateId, String tenantDomain) throws CertificateMgtException {
```

Comment:
why not we make this a method that returns a boolean than throwing an error.

Reply:
Then we have to throw the client exception inside the method where this is invoked.

Reply:
+1 for sending the boolean, don't propagate exceptions to the top unnecessarily

Reply:
This is a private method which is only checked in updateCertificate() method where it checks whether an certificate is existing or not for the given id. 
Based on the boolean response should we ignore updating and proceed, without sending an exception ?
Then how does the parent validate whether the certificate update is successful or not ?

Reply:
The purpose of the method is not really to check the existence, rather to handle the cert unavailability case right ? If so can we rename the method to reflect it's purpose ?

Reply:
or you can just return false and throw the exception from the caller. That will actually have improve the code readability

Reply:
Changed method name.
https://github.com/wso2/carbon-identity-framework/pull/6043/commits/524120310dc908182809894fb03869180f6c4ab8